micro-rdk/src/esp32/encoder.rs

```diff
@@ -57,6 +57,43 @@ pub struct Esp32Encoder<A, B> {
     b: B,
 }
 
+pub struct Esp32EncoderConfig {
+    pub(crate) a: Option<i32>,
+    pub(crate) b: Option<i32>,
+}
+
+use crate::common::config::{AttributeError, Kind};
+impl TryFrom<&Kind> for Esp32EncoderConfig {
+    type Error = AttributeError;
+    fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+        let a = match value.get("a") {
+            Ok(opt) => match opt {
+                Some(val) => Some(val.try_into()?),
+                None => None,
+            },
+            Err(err) => match err {
+                AttributeError::KeyNotFound(_) => None,
+                _ => {
+                    return Err(err);
+                }
+            },
+        };
+        let b = match value.get("b") {
+            Ok(opt) => match opt {
+                Some(val) => Some(val.try_into()?),
+                None => None,
```

Comment:
I think, instead of having `a` and `b` be Options, this case should return an `AttributeError::ValidationError`